app/models/shipit/deploy_spec/npm_discovery.rb

```diff
@@ -79,7 +92,67 @@ def package_version
         JSON.parse(package_json.read)['version']
       end
 
+      def publish_config
+        JSON.parse(package_json.read)['publishConfig']
+      end
+
+      def publish_config_access
+        publish_config['access'] || PRIVATE
+      end
+
+      def package_name
+        JSON.parse(package_json.read)['name']
+      end
+
+      def scoped_package?
+        package_name.start_with?('@shopify')
+      end
+
+      def publish?
+        return false if publish_config.blank?
```

Comment:
Requiring this config for every shipit install is maybe overkill.  Could be make this a per-install option that we'd enable for `shopify/shipit`?

Reply:
> Requiring this config for every shipit install is maybe overkill.

Doesn't this only run on all npm package deploys?  From our conversation in the pod on Friday, I thought we were okay with doing this?

Reply:
If I said that, I've changed my mind ¯\\_(ツ)_/¯  Once publishConfig is fixed, this message is just noise.  

Once `misconfigured-npm-publish-config` displays actionable feedback, we probably have a Good Enough experience.

Reply:
Okay removing `publish_config.blank?` will publish packages privately to PC if they don't have an explicit `publishConfig`. Which is what we wanted the default behaviour to be. I'll add some tests to cover this too. 

--- 
`publish?` will now only test `valid_publish_config_access? && package_scoped_when_private?`

I think we should update the `misconfigured-npm-publish-config` with fixes for these potential misconfigurations

How does this sound?
```
Misconfigured package.json, please ensure the following has been met:

 * If your package is private, ensure your the package name is scoped (@organization/package)
 * Please add a publishConfig with an access setting in your package.json.  e.g.:

"publishConfig:" {
   "access": "<public|restricted>"
}
```

Reply:
Spoke with @GoodForOneFare offline about this

We'd want a way to enforce `publishConfigs` being present in `package.json`'s but without breaking other installs of shipit. We're thinking this can be done with an environment variable that set in shopify's instace of shipit and enforce `publishConfig` with something like:

```
return 'Missing publishConfig in package.json' if ENV['ENFORCE_PUBLISH_CONFIG'] && !has_publish_config?
```

Reply:
Same than for the other question, I'd rather rely on a config than on an env var.